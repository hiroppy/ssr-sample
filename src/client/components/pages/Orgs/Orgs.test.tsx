import * as React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Orgs } from '.';

const match = {
  params: {
    org: 'org'
  },
  isExact: true,
  path: '/',
  url: ''
};

const repos = [
  {
    forksCount: 100,
    name: 'foo',
    url: 'url',
    language: 'lang',
    issuesCount: 200,
    stargazersCount: 300,
    watchersCount: 400
  },
  {
    forksCount: 1000,
    name: 'bar',
    url: 'url',
    language: 'lang',
    issuesCount: 2000,
    stargazersCount: 3000,
    watchersCount: 4000
  }
];

test('should show loader', () => {
  process.env.IS_BROWSER = 'true';

  const mock: any = jest.fn();
  const tree = shallow(
    <Orgs
      error={null}
      name=""
      repos={repos}
      match={match}
      location={mock}
      history={mock}
      load={() => {}}
      isFetchingRepos={true}
    />
  );

  expect(toJson(tree)).toMatchSnapshot();

  delete process.env.IS_BROWSER;
});

test('should call load', () => {
  const mock: any = jest.fn();
  const load = jest.fn();

  mount(
    <Orgs
      error={null}
      name=""
      repos={repos}
      match={match}
      location={mock}
      history={mock}
      load={load}
      isFetchingRepos
    />
  );

  expect(load).toHaveBeenCalled();
});
