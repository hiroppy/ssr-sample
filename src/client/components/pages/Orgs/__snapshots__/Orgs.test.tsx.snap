// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show loader 1`] = `
<branch(branch(OrgsComponent))
  error={null}
  history={[MockFunction]}
  isFetchingRepos={true}
  load={[Function]}
  location={[MockFunction]}
  match={
    Object {
      "isExact": true,
      "params": Object {
        "org": "org",
      },
      "path": "/",
      "url": "",
    }
  }
  name=""
  repos={
    Array [
      Object {
        "forksCount": 100,
        "issuesCount": 200,
        "language": "lang",
        "name": "foo",
        "stargazersCount": 300,
        "url": "url",
        "watchersCount": 400,
      },
      Object {
        "forksCount": 1000,
        "issuesCount": 2000,
        "language": "lang",
        "name": "bar",
        "stargazersCount": 3000,
        "url": "url",
        "watchersCount": 4000,
      },
    ]
  }
  statusCode={null}
/>
`;
